lidarrmetadata/api.py

```
@@ -460,6 +522,81 @@ def search_route():
+    ## we keep cache for different metadata versions separate
+    base_url = app.config['CLOUDFLARE_URL_BASE'] + '/' +  app.config['APPLICATION_ROOT'].lstrip('/').rstrip('/')
+    
+    ## Use a cache key to make sure we don't trigger this in parallel
+    invalidation_in_progress_key = base_url + 'CacheInvalidationInProgress'
+    in_progress = util.CACHE.get(invalidation_in_progress_key)
+    if in_progress:
+        return jsonify('Invalidation already in progress'), 500
+    
+    try:
+        util.CACHE.set(invalidation_in_progress_key, True, timeout=60*5)
+        logger.info('Invalidating cache')
+
+        ## clear cache for all providers, aggregating a list of artists/albums
+        ## that we need to invalidate the final responses for
+        artists = set()
+        albums = set()
+        invalidated = []
+
+        cache_users = provider.get_providers_implementing(provider.InvalidateCacheMixin)
+        for cache_user in cache_users:
+            result = cache_user.invalidate_cache(base_url)
+
+            artists = artists.union(result['artists'])
+            albums = albums.union(result['albums'])
+
+        for artist in artists:
+            util.CACHE.delete_memoized(get_artist_info, artist)
+            util.CACHE.delete_memoized(get_artist_albums, artist)
 
+            key = '{url}/artist/{artist}'.format(url=base_url, artist=artist)
+            invalidated.append(key)
+
+        for album in albums:
+            util.CACHE.delete_memoized(get_release_group_info, album)
+
+            key = '{url}/album/{album}'.format(url=base_url, album=album)
+            invalidated.append(key)
+
+        # cloudflare only accepts 500 files at a time
+        for i in xrange(0, len(invalidated), 500):
+            invalidate_cloudflare(invalidated[i:i+500], retries = 2)
+    
+    finally:
+        util.CACHE.delete(invalidation_in_progress_key)
+        # make sure any exceptions are not swallowed
+        pass
```

Is this necessary? It seems `finally` should only swallow an exception if it contains `return` or `break?